File: 1436. Destination City.cpp
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        unordered_set<string> starting_cities;

        for (auto &path : paths) {
            starting_cities.insert(path[0]);
        };

        unordered_set<string>::const_iterator end = starting_cities.end();
        for (auto &path : paths) {
            if (starting_cities.find(path[1]) == end) {
                return path[1];
            };
        };

        return paths[0][0];
    };
};

File: 1436. Destination City.py
class Solution:
    # O(n) time and space complexity
    def destCity(self, paths: List[List[str]]) -> str:
        cities = set()
        trips = {}

        for trip in paths:
            cities.add(trip[0])
            cities.add(trip[1])
            trips[trip[0]] = trip[1]

        for city in cities:
            if city not in trips:
                return city
        
        return paths[0][0]
